apis/cloudfront/v1alpha1/custom_types.go

```diff
@@ -17,7 +17,9 @@ limitations under the License.
 package v1alpha1
 
 // CustomDistributionParameters includes the custom fields of Distribution.
-type CustomDistributionParameters struct{}
+type CustomDistributionParameters struct {
+	CallerReference *string `json:"callerReference,omitempty"`
```

Comment:
Should we add some documentation stating that this field is for internal use and appropriately defaulted and the `+kubebuilder:validation:Optional` marker for this field?

M understanding is that both for provisioning & import scenarios, users of this API will never have to supply a value, we are adding this to be able to support import scenarios. This could also be only part of the status.

Reply:
I added the kubebuilder marker. As discussed offline, if `callerReference` field is set, I use it during `preCreate`.